### PR TITLE
AHurynovich NodeJS AWS course Task 6 Backend Part

### PR DESCRIPTION
**Pull request by Shagon1k**
## Main task
**TASK 6.1 is implemented.**
[catalogBatchProcess function](https://github.com/Shagon1k/node-js-aws-be/pull/5/files#diff-8cd386e62730151ee84c5b88d9848033a9fb90711d345e882fa921de261a0499)

**TASK 6.2 is implemented.**
[importFileParser function link](https://github.com/Shagon1k/node-js-aws-be/pull/5/files#diff-b0d61f5bc3fd8e5475ca6cdac996b8e22db74817b326c285386199dcbb5789df)
[catalogItemsQueue queue link](https://github.com/Shagon1k/node-js-aws-be/pull/5/files#diff-5a0fbde7d3514c7853d3b671e8fa7d6fdf7da24cff11113d373e506fac3a47c7)

**FrontEnd Application link:** https://d2zvo2vdnqfgz1.cloudfront.net/

_**All evaluation criteria were completed (+ additional ones)**_

## Additional (optional) tasks
**+1 - catalogBatchProcess lambda is covered by unit tests** [link](https://github.com/Shagon1k/node-js-aws-be/pull/5/files#diff-778045a088636c40e6a4190a3976d36bcacdef60a53a8feee3ee626fbbfd101a)

**+1 - set a Filter Policy for SNS createProductTopic in serverless.yml (Create an additional email subscription and distribute messages to different emails depending on the filter for any product attribute)** [See Subscriptions](https://github.com/Shagon1k/node-js-aws-be/pull/5/files#diff-8cd386e62730151ee84c5b88d9848033a9fb90711d345e882fa921de261a0499) [See SNS publishing](https://github.com/Shagon1k/node-js-aws-be/pull/5/files#diff-f68a1833639f8596e918158e95070404bacb326e706c578313da562732652ce9)

### CODE REVIEW

File: src/import-service/serverless.yml [hunk 1]
```diff
@@ -20,6 +20,27 @@ provider:
...
+  environment:
+    SQS_URL:
+      Ref: SQSQueue
+
+resources:
+  Resources:
+    SQSQueue:
+      Type: AWS::SQS::Queue
+      Properties:
+        QueueName: catalogItemsQueue
```
Comment (Mrdev1ce) [inline src/import-service/serverless.yml:36]: According to the architecture presented on the lecture - SQS Queue should be a part of product-service.

image

**Comment by Mrdev1ce** [comment]
I test it with file attached below and it didn't work. Could you check why, please?

A file inside the archive
[dummy copy.csv.zip](https://github.com/Shagon1k/node-js-aws-be/files/5584116/dummy.copy.csv.zip)

File: src/product-service/src/handlers/catalog-batch-process/catalog-batch-process.js [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+	const response = {
+		statusCode: 202,
+		message: responseMsg,
+		data: {
+			addedProducts: JSON.stringify(addedProducts),
+		},
+	};
+
+	return response;
```
Comment (Mrdev1ce) [inline src/product-service/src/handlers/catalog-batch-process/catalog-batch-process.js:100]: catalogBatchProcess is triggered by SQS Event. It should not have response.

**Comment by Shagon1k** [comment]
@Mrdev1ce the reason why your file does not import is that s3 filename is encoded.
image

Thus, we try to get file with a key which does not exists.

**Pull request merged**